### PR TITLE
Revise `experiment` construction to utilize keyword arguments to enhance clarity

### PR DESCRIPTION
**Pull request by navidcy**
This PR changes the experiment constructor, more towards to what #32 is suggesting.

The PR introduces a breaking change in the API. 

### before this PR

```python
import regional_mom6 as rmom6

yextent = [-48, -38.95] ## latitude
xextent = [143, 150] ## longitude
daterange = ["2003-01-01 00:00:00", "2003-01-05 00:00:00"]
inputdir = Path(f"mom6_input_directories/{expt_name}/")
rundir = Path(f"mom6_run_directories/{expt_name}/")
toolpath = Path("PATH_TO_FRE_TOOLS")

expt = rmom6.experiment(
    xextent,
    yextent,
    daterange,
    0.05,  # Horizontal Resolution
    75,    # Number of vertical layers
    10,    # Ratio of largest to smallest vertical layer.
    4500,  # Depth of simulation
    rundir,
    inputdir,
    toolpath
)
```

### after this PR (preliminary, kwarg names to be discussed)

```python
import regional_mom6 as rmom6

expt = rmom6.experiment(
    longitude_extent = [-48, -38.95],
    latitude_extent = [143, 150],
    date_range = ["2003-01-01 00:00:00", "2003-01-05 00:00:00"],
    resolution = 0.05,
    number_vertical_layers = 75,
    layer_thickness_ratio = 10,
    depth = 4500,
    mom_run_dir = Path(f"mom6_input_directories/{expt_name}/"),
    mom_input_dir = Path(f"mom6_run_directories/{expt_name}/"),
    toolpath_dir = Path("PATH_TO_FRE_TOOLS")
)
```

The latter is really self-explanatory; no need to add comments next to every argument to remind people what each arg is.

### CODE REVIEW

**Comment by navidcy** [comment]
Is there a reason that the private method here:

https://github.com/COSIMA/regional-mom6/blob/0143c6405e6253fede7fef467d4132cb6182215f/regional_mom6/regional_mom6.py#L593

takes `grid_type` as an argument and doesn't infer it from `self`?

**Comment by navidcy** [comment]
Another question: 

The docstring here

https://github.com/COSIMA/regional-mom6/blob/0d6f617fa1b5c2fca61518a76933dd5922ca924c/regional_mom6/regional_mom6.py#L520

mentions that only "``xextent``, ``yextent``, ``daterange``, and ``resolution``" are required but I see more args that don't have default values so we need to update it, right?

**Comment by angus-g** [comment]
I don't think we need to repeat the argument list in the description (which also specifies which arguments are optional), and at least `xextent` and `yextent` have been renamed since.

**Comment by navidcy** [comment]
> I don't think we need to repeat the argument list in the description (which also specifies which arguments are optional), and at least `xextent` and `yextent` have been renamed since.

Agree, we can shorten the description!

File: demos/access_om2-forced.ipynb [hunk 1]
```diff
@@ -422,28 +422,26 @@
    "source": [
```
Comment (angus-g) [inline demos/access_om2-forced.ipynb:422]: <pre class="ql-syntax" spellcheck="false">Line #9.    inpu_tdir = f"{scratch}/regional_mom6_configs/{expt_name}/"
</pre><p>oops?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/COSIMA/regional-mom6/pull/113/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (angus-g) [inline demos/access_om2-forced.ipynb:422]: <pre class="ql-syntax" spellcheck="false">Line #21.    for i,orientation in enumerate(["south", "north", "west", "east"]):
</pre><p><code>for i, orientation ...</code> (i.e. add a space, as you did in the reanalysis-forced notebook)?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/COSIMA/regional-mom6/pull/113/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (navidcy) [reply demos/access_om2-forced.ipynb:422]: nice catch!

Comment (navidcy) [reply demos/access_om2-forced.ipynb:422]: done

**Comment by navidcy** [comment]
Do you like the change of `args` to `kwargs` @angus-g?

**Comment by angus-g** [comment]
> Do you like the change of `args` to `kwargs` @angus-g?

I think so, given there are so many arguments (and 4 numbers in a row), it definitely increases clarity.

**Comment by ashjbarnes** [comment]
> Is there a reason that the private method here:
> 
> https://github.com/COSIMA/regional-mom6/blob/0143c6405e6253fede7fef467d4132cb6182215f/regional_mom6/regional_mom6.py#L593
> 
> takes `grid_type` as an argument and doesn't infer it from `self`?

No there's no reason for this actually!

**Review by ashjbarnes — APPROVED** [review]
This looks good. Are the changes still API changing if the arguments remain in the same order now that they're kwargs? If so this should also trigger a new tag release alongside #106

**Comment by navidcy** [comment]
Yeap, we should include this in a new release.
It's a user-facing API change since the users need to provide keywords now; their previous notebook won't work out of the box.

**Pull request merged**